CMakeLists.txt

```diff
@@ -45,55 +45,85 @@ project(viam-cpp-sdk
 
 
 # Configure cmake-level options:
+
+# - `BUILD_SHARED_LIBS`
 #
-# - `BUILD_SHARED_LIBS`: Enabled by default so that we produce a
+# Enabled by default so that we produce a
 # modern SDK, this option can be set to `OFF` to build a static
 # library. Note that this may result in non-functional examples.
 #
 option(BUILD_SHARED_LIBS "If enabled, build shared libraries" ON)
 
 
 # Configure project-global options:
+
+# - `VIAMCPPSDK_ENFORCE_COMPILER_MINIMA`
+#
+# The user can elect to disable enforcement of compiler minima (at
+# their peril).
 #
-# - `VIAMCPPSDK_ENFORCE_COMPILER_MINIMA`: The user can elect to
-# disable enforcement of compiler minima (at their peril).
 option(VIAMCPPSDK_ENFORCE_COMPILER_MINIMA "Control whether we validate the selected compiler against known minima" ON)
 
 
-# # - `VIAMCPPSDK_OFFLINE_PROTO_GENERATION`: Do not use buf.builds remote
-# definitions or services and use the local proto compiler plugins to
-# do generation.
+# - `VIAMCPPSDK_OFFLINE_PROTO_GENERATION`
+#
+# Do not use buf.builds remote definitions or services and use the
+# local proto compiler plugins to do generation.
+#
 option(VIAMCPPSDK_OFFLINE_PROTO_GENERATION "Use local tools for proto generation" OFF)
 
 
-# - `VIAMCPPSDK_USE_DYNAMIC_PROTOS`: The user can select whether to
-# use the prebuilt proto files under the `gen` directory, or to
-# dynamically regenerate them as part of the build. In the latter
-# mode, additional targets are defined that will copy the newly
-# generated protos back to the source directory so they can be
-# committed:
+# - `VIAMCPPSDK_USE_DYNAMIC_PROTOS`
+#
+# The user can select whether to use the prebuilt proto files under
+# the `gen` directory, or to dynamically regenerate them as part of
+# the build. In the latter mode, additional targets are defined that
+# will copy the newly generated protos back to the source directory so
+# they can be committed:
 #
 # ```
 # cmake ... -DVIAMCPPSDK_USE_DYNAMIC_PROTOS=ON
 # [make|ninja|...] update-static-protos
 # ```
 # Remember to check-in the results after regenerating!
+#
 option(VIAMCPPSDK_USE_DYNAMIC_PROTOS "Regenerate protos as part of building (network access may be required)" OFF)
 
 
-# # - `VIAMCPPSDK_USE_WALL_WERROR`: This causes the SDK's internal
-# code to compile with `-Wall` and `-Werror` flags.
+# - `VIAMCPPSDK_USE_LOCAL_PROTOS`
+#
+# Builds against local proto definitions, as opposed to those in the
+# buf registry. This flag assumes that there is a `buf.workspace` file
```

Comment:
@acmorrow I know you're generally opposed to `.gitignore` proliferation, but I wonder if it's worth adding `buf.work.yaml` to the `.gitignore` in this case. I'd be surprised if we ever want to check one in, but I could certainly imagine that other repos would want to check one in such that adding it to a user's `.gitignore` is a little bit annoying.

Reply:
The `buf.work.yaml` file goes in the parent directory of the SDK, so there isn't anything to git ignore.

Reply:
Ah, so it is! Disregard :)